app/org/sagebionetworks/bridge/services/SurveyService.java

```diff
@@ -169,11 +181,24 @@ public void deleteSurveyPermanently(StudyIdentifier studyId, GuidCreatedOnVersio
         checkArgument(StringUtils.isNotBlank(keys.getGuid()), "Survey GUID cannot be null/blank");
         checkArgument(keys.getCreatedOn() != 0L, "Survey createdOn timestamp cannot be 0");
 
+        // verify shared module existence as well
+        verifySharedModuleExistence(keys);
+
```

Comment:
I think you misunderstood. I was suggesting moving it *into* checkConstraintsBeforePhysicalDelete(), not change the order.